### PR TITLE
only pass $id and $email if available; remove fallback to anonymousId

### PR DESCRIPTION
**Pull request by WesleyDRobinson**

### CODE REVIEW

File: lib/mapper.js [hunk 1]
```diff
@@ -53,15 +53,20 @@ exports.identify = function(identify){
  */
 
 exports.track = function(track){
+  var customer_properties = {};
+  var userId = track.userId();
+  var email = track.email();
+
+  // at least one of userId or email is required by Klaviyo
+  if (userId) customer_properties.$id = userId;
```
Comment (hankim813) [inline lib/mapper.js:61]: we don't need `userId` or `email` variables. I would just use `track.userId()`

File: lib/mapper.js [hunk 2]
```diff
@@ -53,15 +53,20 @@ exports.identify = function(identify){
...
+  if (email) customer_properties.$email = email;
```
Comment (hankim813) [inline lib/mapper.js:62]: same here

**Review by hankim813 — CHANGES_REQUESTED** [review]
Could you also write unit tests to make sure we test setting just userId or email and both?

Also could you write a summary of the decision made from klaviyo and our end in the PR description so we can link back for reference in the future? 

Also helpful things to include is the jira ticket if there is one.

File: lib/mapper.js [hunk 3]
```diff
@@ -53,15 +53,20 @@ exports.identify = function(identify){
...
+
   return {
     token: this.settings.apiKey,
     event: track.event(),
     properties: properties(track),
     time: time(track.timestamp()),
-    customer_properties: {
-      $id: track.userId() || track.sessionId(),
```
Comment (hankim813) [inline lib/mapper.js:62]: what's the reason for removing the fallback to `anonymousId` here?

Comment (WesleyDRobinson) [reply lib/mapper.js:62]: Passing a $id to Klaviyo overrides their unique-email-address enforcement so many accounts are being created tied to a single email address when we fallback to anonymousId

Comment (WesleyDRobinson) [reply lib/mapper.js:62]: logic update needs to be applied to all ecommerce events

File: lib/mapper.js (hunk 1 above)
Comment (WesleyDRobinson) [reply lib/mapper.js:61]: I can't find docs or examples for how the mapper should behave when encountering this kind of failure. Should it throw an error up for .mapper() to handle?

**Pull request closed without merging**